IronGram/src/main/java/com/theironyard/controllers/IronGramController.java

```diff
@@ -101,13 +103,28 @@ public Photo upload(HttpSession session, MultipartFile photo,
             dir.mkdirs();
         }
 
+        if (!photo.getContentType().startsWith("image")){
+            throw new FileNotAnImage();
+        }
+
         File photoFile = File.createTempFile("photo", photo.getOriginalFilename(), dir);
         FileOutputStream fos = new FileOutputStream(photoFile);
         fos.write(photo.getBytes());
-
         Photo p = new Photo(sender, receiver, photoFile.getName());
+        if(publicPhoto == null){
```

Comment:
This might be a good a good spot for a one line if statement like `isPublic = (publicPhoto == null ? true : false);`